### PR TITLE
Pull request for lesson 4 assignments

### PR DESCRIPTION
**Pull request by bplanica**
Pull request for lesson 4 assignments

Thanks! :)

Breeanna

### CODE REVIEW

File: students/bplanica/lesson04/mailroom2.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+#                   Adding function to print emails to text files
+# ------------------------------ #
+
+# ----- DATA ----- #
+# ---------------- #
+donors = [{"Donor": "Bill Gates", "Donation": [326892.24, 326892.24]},
+{"Donor": "Mark Zuckerberg", "Donation": [5465.39, 5465.39, 5465.39]},
+{"Donor": "Jeff Bezo", "Donation": [877.33]},
+{"Donor": "Paul Allen", "Donation": [236.14, 236.14, 236.14]},
+{"Donor": "M Bezo", "Donation": [110463.25]}] # holds original data set to start with
```
Comment (charlesrobison) [inline students/bplanica/lesson04/mailroom2.py]: This works, but it's a list containing dictionaries.  Ideally, you'd want a single dictionary with the donors as keys.  For example:

`def get_donor_db():
    return {'bill gates': ("Bill Gates", [326892.24, 326992.24]),
            'mark zuckerberg': ("Mark Zuckerberg", [5465.39, 5465.39, 5465.39]),
            'etc, etc,
            }`

File: students/bplanica/lesson04/mailroom2.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+def quit_program():
+    raise SystemExit # exits the script all together
+
+
+# ----- PRESENTATION ----- #
+# ------------------------ #
+if __name__ == '__main__':
+    dict_menu = {1: one_thank_you, 2: create_report,
+        3: all_thank_you, 4: quit_program} # holds menu options (dispatch)
+    main() # run main
```
Comment (charlesrobison) [inline students/bplanica/lesson04/mailroom2.py]: Works well overall - nice job

File: students/bplanica/lesson04/trigrams.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+# ----- Presentation ----- #
+# ------------------------ #
+if __name__ == "__main__":
+    lst_words = read_file(file_name) # gather the list of words
+    trigrams = build_trigrams(lst_words) # build the dictionary
+    #for key, value in trigrams.items():
+    #    print(f"{key}, {value}")
+    lst_story = build_story(trigrams) # build the story
+    print(" ".join(lst_story)) # print the list/story with spaces in between words
```
Comment (charlesrobison) [inline students/bplanica/lesson04/trigrams.py]: Tried this with a few files and keep getting a list index out of range error.  Does it run for you?  Can you provide the text file and outline the steps for a successful run?

**Comment by natasha-aleksandrova** [comment]
@charlesrobison can you review trigrams again please?

**Comment by charlesrobison** [comment]
Hi @bplanica - let's keep communication either in the PR, or via the instructor email alias - keeps things simpler ;)

There's no need to re-submit work on the course site, you can just do a new PR and we can go in, review the PR, and adjust grades if needed.  So, please continue to submit work on the course site at the same time you're doing your initial PR.  Does that make sense?


Charles

**Pull request merged**